### PR TITLE
fix @GraphQLConnection for [ObjectType!]!

### PR DESCRIPTION
**Pull request by asinbow**
For easing the frontend development in strict typed system like `typescript` or `flowjs`, we usually use `double @GraphQLNonNull like [ObjectType!]!`. But there is a small bug in `ConnectionUtil.java` that it doesn't unwrap the internal NonNull`!`. This PR is for it , thanks for reviewing.

### CODE REVIEW

**Comment by asinbow** [comment]
~~Totally new about this repository.
I want to improve unit test for it, any suggestion?~~

Added my tests for it. I think it's ready for reviewing.

File: src/main/java/graphql/annotations/processor/util/ConnectionUtil.java [hunk 1]
```diff
@@ -33,13 +33,24 @@
...
+
+        if (type instanceof graphql.schema.GraphQLNonNull) {
+            type =  ((GraphQLNonNull) type).getWrappedType();
+        }
+        if (!(type instanceof GraphQLList)) {
+          return false;
+        }
+
+        type = ((GraphQLList) type).getWrappedType();
         if (type instanceof graphql.schema.GraphQLNonNull) {
```
Comment (asinbow) [inline src/main/java/graphql/annotations/processor/util/ConnectionUtil.java:48]: An extra unwrap for types like `[ObjectType!]!`.

File: src/test/java/graphql/annotations/connection/GraphQLConnectionTest.java [hunk 2]
```diff
@@ -201,6 +207,19 @@ public void methodNonNull() {
         testResult("getNonNullObjs", result);
     }
 
+    @Test
+    public void methodDoubleNonNull() {
```
Comment (asinbow) [inline src/test/java/graphql/annotations/connection/GraphQLConnectionTest.java:211]: test double `@GraphQLNonNull` like `[ObjectType!]!`.

File: src/test/java/graphql/annotations/connection/GraphQLConnectionTest.java [hunk 3]
```diff
@@ -396,4 +422,42 @@ public String getCursor(Obj entity) {
 
     }
 
+    /*
+     * double @GraphQLNonNull on @GraphQLList like [ObjectType!]!
+     */
+    public static class DoubleNonNullIterableTypeFunction implements TypeFunction {
```
Comment (asinbow) [inline src/test/java/graphql/annotations/connection/GraphQLConnectionTest.java:428]: As I have experimented, this is the best way to implement `double @GraphQLNonNull like [ObjectType!]!`.
How about to make it builtin?

Comment (yarinvak) [reply src/test/java/graphql/annotations/connection/GraphQLConnectionTest.java:428]: interesting.. we have to think about it (I don't think we handle double graphql non-null at all, not only in connection)

**Review by yarinvak — APPROVED** [review]

**Pull request merged**